hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionRegistry.java

```diff
@@ -18,21 +18,36 @@
 package org.apache.hadoop.hbase.client;
 
 import java.io.Closeable;
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import org.apache.hadoop.hbase.HRegionLocation;
+import org.apache.hadoop.hbase.RegionLocations;
 import org.apache.hadoop.hbase.ServerName;
 import org.apache.yetus.audience.InterfaceAudience;
 
 /**
  * Registry for meta information needed for connection setup to a HBase cluster. Implementations
- * hold cluster information such as this cluster's id, location of hbase:meta, etc..
- * Internal use only.
+ * hold cluster information such as this cluster's id, location of hbase:meta, etc.. Internal use
+ * only.
  */
 @InterfaceAudience.Private
 interface ConnectionRegistry extends Closeable {
 
+  /**
+   * Get location of meta region for the given {@code row}.
+   */
+  CompletableFuture<RegionLocations> locateMeta(byte[] row, RegionLocateType locateType);
+
+  /**
+   * Get all meta region locations, including the location of secondary regions.
+   * @param excludeOfflinedSplitParents whether to include split parent.
+   */
+  CompletableFuture<List<HRegionLocation>>
```

Comment:
I think it is fine? The locateMeta method will be called much more than getAllMetaRegionLocations, if we merge them into one method, the parameters will be more complicated.

Reply:
I think it's fine to keep them separate, logically it might make sense to combine them but given that Rpc service interface also has the same separation, here we can maintain it for easy mapping. Yes it is 1:1 with RPC call, but still looks simple. Anyways, I don't have strong preference here.